plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.ts

```diff
@@ -95,6 +95,35 @@ export class StackOverflowQuestionsCollatorFactory
     return Readable.from(this.execute());
   }
 
+  async getQuestions() {
+    let hasMorePages = true;
+    let page = 1;
+    const items = [];
+
+    const params = qs.stringify(this.requestParams, {
+      arrayFormat: 'comma',
+      addQueryPrefix: true,
+    });
+
+    const apiKeyParam = this.apiKey
+      ? `${params ? '&' : '?'}key=${this.apiKey}`
+      : '';
+
+    while (hasMorePages) {
+      const res = await fetch(
+        `${this.baseUrl}/questions${params}${apiKeyParam}&page=${page}`,
+      );
+
+      const data = await res.json();
+      items.push(...data.items);
+      hasMorePages = data.has_more;
+      page = page + 1;
+    }
+    return {
+      items,
+    };
```

Comment:
Possibly `#premature-optimization` (so consider this optional), but if you keep this looping logic in the main `*execute()` method and `yield` each result, you might get an overall performance improvement in the indexing process:

Yielding questions as they're loaded may reduce the overall memory usage (won't have to store all questions in memory on the `items` variable), and could also reduce indexing time (because decorators/indexer might get their hands on the document while this is still retrieving pages of questions from the API).

Reply:
Nice feedback, I'll update to that!

Reply:
updated here https://github.com/backstage/backstage/pull/13253/commits/0df9a2639fa262aa95dabcbe1cfb720fca79cb7e